homeassistant/components/honeywell/climate.py

```diff
@@ -106,70 +108,58 @@
-    password = config.get(CONF_PASSWORD)
-
-    try:
-        client = somecomfort.SomeComfort(username, password)
-    except somecomfort.AuthError:
-        _LOGGER.error("Failed to login to honeywell account %s", username)
-        return
-    except somecomfort.SomeComfortError:
-        _LOGGER.error(
-            "Failed to initialize the Honeywell client: "
-            "Check your configuration (username, password), "
-            "or maybe you have exceeded the API rate limit?"
-        )
-        return
-
-    dev_id = config.get(CONF_DEV_ID)
-    loc_id = config.get(CONF_LOC_ID)
-    cool_away_temp = config.get(CONF_COOL_AWAY_TEMPERATURE)
-    heat_away_temp = config.get(CONF_HEAT_AWAY_TEMPERATURE)
-
-    add_entities(
-        [
-            HoneywellUSThermostat(
-                client,
-                device,
-                cool_away_temp,
-                heat_away_temp,
-                username,
-                password,
-            )
-            for location in client.locations_by_id.values()
-            for device in location.devices_by_id.values()
-            if (
-                (not loc_id or location.locationid == loc_id)
-                and (not dev_id or device.deviceid == dev_id)
+    cool_away_temp = config.data.get(CONF_COOL_AWAY_TEMPERATURE)
+    heat_away_temp = config.data.get(CONF_HEAT_AWAY_TEMPERATURE)
+
+    data = hass.data[DOMAIN][config.entry_id]
+
+    async_add_entities([HoneywellUSThermostat(data, cool_away_temp, heat_away_temp)])
+
+    return True
+
+
+async def async_setup_platform(hass, config, add_entities, discovery_info=None):
+    """Honeywell uses config flow for configuration now. If an entry exists in configuration.yaml, the import flow will attempt to import it and create a config entry."""
+
+    if config["platform"] == "honeywell":
+        _LOGGER.debug("Deprecated honeywell entry found in configuration.yaml")
+        # No config entry exists and configuration.yaml config exists, trigger the import flow.
+        if not hass.config_entries.async_entries(DOMAIN):
+            await hass.config_entries.flow.async_init(
+                DOMAIN, context={"source": SOURCE_IMPORT}, data=config
             )
-        ]
-    )
+
+    return True
 
 
 class HoneywellUSThermostat(ClimateEntity):
     """Representation of a Honeywell US Thermostat."""
 
-    def __init__(
-        self, client, device, cool_away_temp, heat_away_temp, username, password
-    ):
+    def __init__(self, data, cool_away_temp, heat_away_temp):
         """Initialize the thermostat."""
-        self._client = client
-        self._device = device
+        self._data = data
         self._cool_away_temp = cool_away_temp
         self._heat_away_temp = heat_away_temp
         self._away = False
-        self._username = username
-        self._password = password
 
-        _LOGGER.debug("latestData = %s ", device._data)
+        self._unique_id = homeassistant.helpers.device_registry.format_mac(
```

Comment:
```suggestion
        self._attr_unique_id = homeassistant.helpers.device_registry.format_mac(
```

Do the same with other constant proeprties, e.g. for example `name`

Reply:
How would HA know the unique_id of this thermostat if I do this? Is `self._attr_<name>` equivalent to `@property`?

Reply:
> Is self._attr_<name> equivalent to @property

Yes, we've very recently introduced this as a way to decrease the amount of boilerplate code for integrations 👍
This is used in many places already, you might have branched out from a too old commit. Please rebase if that's the case.

Reply:
Nice, that did clean out a lot of code. Thanks for the tip! 👍